apps/astarte_realm_management/test/astarte_realm_management/new_engine_test.exs

```diff
@@ -213,6 +217,59 @@ defmodule Astarte.RealmManagement.EngineTestv2 do
     end
   end
 
+  describe "Test triggers" do
+    @describetag :triggers
+    property "are installed correctly", %{realm: realm} do
+      check all(
+              interface <- Astarte.Core.Generators.Interface.interface(),
+              device <- Astarte.Core.Generators.Device.device(interfaces: [interface]),
+              trigger <- trigger(string(:utf8)),
+              policy <- Astarte.Core.Generators.Triggers.Policy.policy(),
+              simple_trigger <-
+                simple_trigger_config(interface.name, interface.major_version, device.device_id)
+            ) do
+        _ = Engine.install_trigger_policy(realm, Jason.encode!(policy))
+
+        :ok =
+          Engine.install_trigger(
+            realm,
+            trigger.name,
+            policy.name,
+            trigger.action,
+            serialize_simple_triggers([simple_trigger])
+          )
+
+        {:ok,
+         %{
+           trigger: fetched_trigger,
+           serialized_tagged_simple_triggers: serialized_tagged_simple_triggers
+         }} =
+          Engine.get_trigger(realm, trigger.name)
+
+        assert trigger.action == fetched_trigger.action
+        assert trigger.name == fetched_trigger.name
+        assert trigger.version == fetched_trigger.version
+        assert policy.name == fetched_trigger.policy
+
+        serialized_tagged_simple_triggers =
+          serialized_tagged_simple_triggers
+          |> Enum.map(fn stst ->
+            TaggedSimpleTrigger.decode(stst)
+            |> SimpleTriggerConfig.from_tagged_simple_trigger()
+          end)
+
+        simple_trigger =
+          unless simple_trigger.interface_major,
+            do: Map.put(simple_trigger, :interface_major, 0),
+            else: simple_trigger
+
+        assert serialized_tagged_simple_triggers == [simple_trigger]
+
+        _ = Engine.delete_trigger(realm, trigger.name)
```

Comment:
This would fit better in the `on_exit` callback:
```suggestion
        on_exit(fn -> Engine.delete_trigger(realm, trigger.name) end)
```

Reply:
With an `on_exit` the function is called only after the whole property is tested, while that call has to be made at each `check`. The generated triggers might have the same name, so at the end of each run if there are no errors is better to delete the newly installed trigger.

Reply:
OK!